### PR TITLE
Prevent an edge case notice, and a unit test

### PR DESCRIPTION
**Pull request by kienstra**
Once, I think when importing a block, this notice appeared:
>Notice: Trying to get property 'post_type' of non-object in /path/to/wp-content/plugins/block-lab/php/blocks/class-loader.php on line 183

This looks to be from `get_post()` returning something like `null`, and trying to access the `->post_type` property of that.

This is probably an edge case, I didn't see it before.

Sorry, I don't remember the steps to reproduce this.

#### Changes
* Handles a case of the global `$post` being `null`, or not having a `->post_type`
* Adds a basic test

### CODE REVIEW

File: php/blocks/class-loader.php
Comment (kienstra) [inline php/blocks/class-loader.php:168]: This is simply moved below, so the script-related logic is first, then the style logic. There's no change to it.

File: php/blocks/class-loader.php
Comment (kienstra) [inline php/blocks/class-loader.php:168]: This is the main edit in the PR. `get_post_type()` will return `false` if `get_post()` returns something falsey. This should prevent a PHP notice.

**Comment by kienstra** [comment]
Travis should pass once we merge #502 and merge in the `develop` branch to this.

**Comment by kienstra** [comment]
A review can wait until #502 is merged.

File: php/blocks/class-loader.php
Comment (kienstra) [inline php/blocks/class-loader.php:135]: This isn't related to this PR, it's just something I noticed could be more concise.

**Review by lukecarbis — APPROVED** [review]

**Pull request merged**
